packages/server/graphql/public/types/Team.ts

```diff
@@ -52,15 +53,13 @@ const Team: TeamResolvers = {
     const teamMembers = await dataLoader.get('teamMembersByTeamId').load(teamId)
     return teamMembers.find((teamMember) => teamMember.isLead)!
   },
-  isViewerTeamLead: async ({id: teamId}, _args, {authToken, dataLoader}) => {
-    const viewerId = getUserId(authToken)
-    const teamMemberId = toTeamMemberId(teamId, viewerId)
-    const teamMember = await dataLoader.get('teamMembers').load(teamMemberId)
-    return teamMember?.isLead || false
-  },
   insight: async ({id: teamId}, _args, {dataLoader}) => {
     const insight = await dataLoader.get('latestInsightByTeamId').load(teamId)
-    return insight || null
+    if (!insight) return null
+    return {
+      ...insight,
+      id: InsightId.join(teamId, insight.id)
```

Comment:
+1 The nicer solution would be to have a resolver for the insights type. Define the DB type as source for it and then have a single id field doing this conversion. This way it will work in all occurrences where we would add this field.